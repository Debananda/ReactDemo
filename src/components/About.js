import React, {Component} from "react";
import {Button} from "primereact/button";

class About extends Component {
    goBack = () => {
        this.props.history.goBack();
    };

    render() {
        return (
            <div>
                <h1>About Component</h1>
                <Button color="primary" onClick={this.goBack} label={"Back"}/>
            </div>
        );
    }
}

export default About;
