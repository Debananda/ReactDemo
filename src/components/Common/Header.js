import React from "react";
import {
    Menubar
} from "primereact/menubar";
import {withRouter} from 'react-router-dom'

const Header = (props) => {
    const navigateToPage = (path) => {
        props.history.push(path)
    };
    const items = [
        {
            label: 'Home', command: () => {
                navigateToPage('/')
            }
        },
        {
            label: 'Posts', command: () => {
                navigateToPage('/posts')
            }
        },
        {
            label: 'About', command: () => {
                navigateToPage('/about')
            }
        }
    ];
    return (
        <Menubar model={items}/>
    );
};

export default withRouter(Header)
