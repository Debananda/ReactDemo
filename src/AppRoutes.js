import React from "react";
import { Switch, Route } from "react-router-dom";
import Home from "./components/Home";
import NotFound from "./components/Common/NotFound";
import Posts from "./components/Posts";
import About from "./components/About";

export default function AppRoutes() {
  return (
    <Switch>
      <Route path="/" exact component={Home} />
      <Route path="/posts" component={Posts} />
      <Route path="/about" component={About} />
      <Route component={NotFound} />
    </Switch>
  );
}
