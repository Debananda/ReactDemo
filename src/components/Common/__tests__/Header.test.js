import React from "react";
import { mount } from "enzyme";
import Header from "../Header";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import AppRoutes from "../../../AppRoutes";
import About from "../../about";

describe("header component testing", () => {
  test("shall render  without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Header />
        <AppRoutes />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  test("shall render  about component on about link click", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Header />
        <AppRoutes />
      </MemoryRouter>
    );
    wrapper
      .find(Header)
      .find("[href='/about']")
      .simulate("click", { button: 0 });
    expect(wrapper.find(About)).toHaveLength(1);
  });
});
