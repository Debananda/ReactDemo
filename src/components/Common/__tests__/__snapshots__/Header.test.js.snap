// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component testing shall have correct snapShot 1`] = `
<div
  className="p-menubar p-component"
  id={null}
  style={null}
>
  <ul
    className="p-menubar-root-list"
  >
    <li
      className="p-menuitem"
      onMouseEnter={[Function]}
    >
      <a
        className="p-menuitem-link"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          className="p-menuitem-text"
        >
          Home
        </span>
      </a>
    </li>
    <li
      className="p-menuitem"
      onMouseEnter={[Function]}
    >
      <a
        className="p-menuitem-link"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          className="p-menuitem-text"
        >
          Posts
        </span>
      </a>
    </li>
    <li
      className="p-menuitem"
      onMouseEnter={[Function]}
    >
      <a
        className="p-menuitem-link"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          className="p-menuitem-text"
        >
          About
        </span>
      </a>
    </li>
  </ul>
</div>
`;
